main.c

```diff
@@ -30,6 +30,37 @@
 #include "wlr-screencopy-unstable-v1-client-protocol.h"
 #include "ext-session-lock-v1-client-protocol.h"
 
+#define MAX_OUTPUT_SIZE 1024
+
+static struct swaylock_state state;
+
+void update_script() {
+	FILE *pipe;
+	char path[1035];
+	char output[MAX_OUTPUT_SIZE];
+	char *script_str = (char *)malloc((MAX_OUTPUT_SIZE + 1) * sizeof(char));
```

Comment:
at the end of the day we're only showing a single line (I'm assuming the intent is to read only one line of the script's output) of `MAX_OUTPUT_SIZE` bytes to the user, so why not just `fgets` that last line into script_str? do we need to be doing all that?